enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/EnterpriseCompatibilityFactory.scala

```diff
@@ -44,8 +47,15 @@ class EnterpriseCompatibilityFactory(inner: CompatibilityFactory, graph: GraphDa
       case (CypherPlanner.rule, _) => inner.create(spec, config)
 
       case _ =>
+        val settings = graph.getDependencyResolver.resolveDependency(classOf[Config])
+        val morselSize = settings.get( GraphDatabaseSettings.cypher_morsel_size)
+        val workers = settings.get( GraphDatabaseSettings.cypher_worker_count)
+        val dispatcher =
+          if (workers == 1) new SingleThreadedExecutor(morselSize)
+          else if (workers == 0) new ForkJoinPoolExecutor(morselSize, Runtime.getRuntime.availableProcessors() * 2)
```

Comment:
Would we be able to have cypher get an executor from https://github.com/neo4j/neo4j/blob/3.4/community/common/src/main/java/org/neo4j/scheduler/JobScheduler.java#L73 ? 

I know it's just moving the same behavior elsewhere, but it seems useful to try to keep thread creation in a single place to ease future expansion of control of jobs.

Note that cypher can still get it's own pool just like this one, it just would get it from the central JobScheduler, see `BoltKernelExtension` for how bolt does this (eg. it depends on getting an Executor or whatever, and just uses the JobScheduler to instantiate one)

Reply:
Why * 2? Runtime.getRuntime.availableProcessors() should include hyper threads so doubling the number of workers will just give us more context switching?

Reply:
You are correct. Don't know what a good value is. Our very small benchmarks do not show a performance drop when we have too many workers, but that does not mean we should. We should probably change this to what you say, but I would love to investigate and have numbers to show us.

Reply:
@jakewins Our JobScheduler does not support bounded threadpools, from what I can see. I think it's important to be able to cap the number of worker threads we use for this. Do you still think we should use `Neo4jJobScheduler` for this?

Reply:
@systay There is `threadFactory(..)`, which lets you be in charge of creating threads; however, since you're using ForkJoinPool, it looks like you can't create a ForkJoinPool from a raw ThreadFactory; you'd need to amend the `JobScheduler` with a `forkJoinThreadFactory()` or `forkJoinThreadPool(int limit)` or some such.

So: Yes, you're right :( But! If you refactor `JobScheduler` to support bounded ForkJoinPools, then others that want to use ForkJoin things can benefit :) You can do that either by replicating the pattern used by `threadFactory()`, by `executor()`, or by amending the `Group` definition to build in rules - eg. "group `CypherWorkers` can only have N workers". The latter would be closer in line with long-term plans; eg. the `JobScheduler` decides how to divide up resources.

By using the JobScheduler for this, we retain a single view of thread creation in the DB - and all these Cypher threads end up following the same naming scheme in thread dumps, which feels good to my OCD.